lib/common/multithread.c

```diff
@@ -87,11 +87,16 @@ static void init_async(h2o_multithread_queue_t *queue, h2o_loop_t *loop)
 {
     int fds[2];
 
+#ifndef _WIN32
     if (cloexec_pipe(fds) != 0) {
         perror("pipe");
         abort();
     }
     fcntl(fds[1], F_SETFL, O_NONBLOCK);
+#else
+    u_long nonblock = 1;
```

Comment:
> Yes I did this yesterday evening using the socket function and making it a stream socket with ipv4 support and tcp but it didn't change much in that the in ev-loop mode example-httpclient hangs (still stuck in the loop)

To which part of the callbacks does the client proceed?  exmaples/libh2o/http1client.c defines a number of callbacks that are called one by one.  `on_connect` is called first, then `on_head`, and `on_body`.  You could also set breakpoints to (or emit logs in) the callback functions within socketpool.c or http1client.c to determine which callbacks are called, and which are not.

That would help you analyze the cause of the issue.

> Should I focus on the evloop version first or the libuv ?

That depends on what you want to do.

If your intention is to use libh2o, then you should better concentrate on libh2o and the examples under examples/libh2o.  OTOH if your intention is porting the standalone server to Windows, I would suggest concentrating on evloop and then working on the libuv-side.

Personally I did not expect somebody to start porting the standalone server to Windows, and am fascinated by how far you have achieved so far.